src/Microsoft.PowerApps.TestEngine/PowerFx/PowerFxHelper.cs

```diff
@@ -16,36 +16,54 @@ public static class PowerFxHelper
         /// Operator is ";" when decimal separator for the locale is "." and is ";;" when decimal separator is ","
         /// </summary>
         /// <param name="engine">Instance of an engine configured with the desired locale</param>
-        /// <param name="expression">Expression from which formulas separated by chaining operator would be extracted</param>
+        /// <param name="result">Check result instance created after processing the expression</param>
         /// <returns>An enumerable of formulas extracted from the expression that are separated by chaining operator</returns>
-        public static IEnumerable<string> ExtractFormulasSeparatedByChainingOperator(Engine engine, string expression)
+        public static IEnumerable<string> ExtractFormulasSeparatedByChainingOperator(Engine engine, CheckResult result)
         {
-            if (string.IsNullOrWhiteSpace(expression))
+            if (string.IsNullOrEmpty(result?.Parse?.Text))
             {
                 return new string[0];
             }
 
+            var spansForFormulasSeparatedAcrossMultipleDepths = ExtractSpansOfFormulasSeparatedByChainingOperator(engine, result?.Parse.Text);
+            if (result?.Parse?.Root == null)
```

Comment:
Maybe add a comment before this if condition?

Reply:
i mean it is pretty straightforward that if the parse tree's root is null then we have to return spans for formulas that are separated at any level. If it is something wasn't clear to you then i can add it. don't wan to leave code smell if it is clear. let me know if you still want me to add it